### ISSUE TITLE
Typescript IntelliSense hang when writing generic types

### ISSUE THREAD

**Issue by indooorsman** [microsoft/vscode#147083]
Issue Type: <b>Bug</b>

everytime I'm writing **generic** types, typescript server seems to hang up and IntelliSense keep loading, and there's a Electron process keep 100% cup usage. this issue happend for me with both stable & insiders build of vscode. here's my `tsconfig.json` & a gif of steps to reproduce & screenshot of process explorer:
```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "preserveSymlinks": true,
    "module": "ESNext",
    "target": "ESNext",
    "lib": [
      "ESNext",
      "DOM",
      "DOM.Iterable"
    ],
    "moduleResolution": "Node",
    "jsx": "react-jsx",
    "isolatedModules": false,
    "noEmit": true,
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "sourceMap": false,
    "baseUrl": ".",
    "rootDir": ".",
    "paths": {
      "@lib/*": [
        "src/lib/*"
      ],
      "@components/*": [
        "src/components/*"
      ]
    },
    "typeRoots": [
      "node_modules/@advertising/types",
      "node_modules/@types"
    ]
  },
  "include": [
    "./node_modules/@advertising/types/*.d.ts",
    "./src/**/*"
  ]
}

```
![tsissue](https://user-images.githubusercontent.com/4919405/162446972-b079a9a6-3f12-4090-9727-1924a95b7a19.gif)
<img width="800" alt="screenshot" src="https://user-images.githubusercontent.com/4919405/162447027-35ce643d-ffb5-44fa-8cf8-6fd06c5d41a1.png">

VS Code version: Code - Insiders 1.67.0-insider (6ea335e334b90a9aca753d558dec1d14741eff1e, 2022-04-08T05:18:35.034Z)
OS version: Darwin x64 21.4.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 3, 3|
|Memory (System)|32.00GB (2.14GB free)|
|Process Argv|--crash-reporter-id dcdd842f-44a5-4728-b1cd-55801a0fcfae|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes516:30244333
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vsdfh931:30280409
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30404738
wslgetstarted:30449409
pythonvsnew555:30438690
vscscmwlcmt:30465136
cppdebug:30466689
pynewfile477:30451556

```

</details>

**Comment by indooorsman** [comment]
@mjbvz could you please help with it? I'm struggling for this issue the whole day 😭

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by mjbvz** [comment]
If it still does, can you please share a minimal sample project so we can try reproducing this?

**Comment by indooorsman** [comment]
> Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

Yes it still reproduce in my project(has many workspaces & packages), but when I try in a minimal sample project it can't reproduce... 😵‍💫
